### PR TITLE
Initial Kube Method Implementation

### PR DESCRIPTION
**Pull request by josephsawaya**
With this method a user can apply multiple multi-document
yaml files specifying config maps and pods.

This handles cases where podman fails, like when the name
of a pod and one of its containers is the same.

This also handles detecting any pods previously deployed
with the same name as one that it is currently trying to
deploy and tears it down.

Signed-off-by: Joseph Sawaya <jsawaya@redhat.com>

### CODE REVIEW

File: cmd/harpoon/main.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func (repo *Repo) process() {
 		// ... get the files iterator and print the file
 		// .. make sure we're only calling the engine method on json files
 		subDirTree.Files().ForEach(func(f *object.File) error {
-			if strings.HasSuffix(f.Name, ".json") || strings.HasSuffix(f.Name, ".service") {
+			if strings.HasSuffix(f.Name, ".json") || strings.HasSuffix(f.Name, ".service") || strings.HasSuffix(f.Name, ".yaml") || strings.HasSuffix(f.Name, ".yaml") {
```
Comment (cooktheryan) [inline cmd/harpoon/main.go:101]: do we want to make this .yml?

**Comment by cooktheryan** [comment]
FYI @sallyom 

I'm good with this to merge

@josephsawaya want to take a run at doing the GitHub action  in a follow up or?

**Comment by josephsawaya** [comment]
> @josephsawaya want to take a run at doing the GitHub action in a follow up or?

sure

File: pkg/engine/kube.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	var i interface{}
+	var pod = v1.Pod{}
+	d := yaml.NewDecoder(bytes.NewReader(kubeYaml))
+
+	pod_map := make(map[string]bool)
+
+	for {
+		err = d.Decode(&i)
+		if err == io.EOF {
```
Comment (jeffvance) [inline pkg/engine/kube.go:42]: `switch` on `err` better? More expandable for future error handling?

**Pull request merged**